lib/include/openamp/rpmsg_virtio.h

```diff
@@ -191,7 +197,7 @@ int rpmsg_init_vdev_with_config(struct rpmsg_virtio_device *rvdev,
 				struct virtio_device *vdev,
 				rpmsg_ns_bind_cb ns_bind_cb,
 				struct metal_io_region *shm_io,
-				struct rpmsg_virtio_shm_pool *shpool,
+				struct rpmsg_virtio_shm_pool **shpool,
```

Comment:
We should avoid to modify API as impact all users.
I was expecting to keep `rpmsg_virtio_shm_pool *`  pointer which points to the first element and use `shpool++` to internally address next array element . 
In addition I quite surprised that you don't need also to duplicate the `shm_io`

An alternative is to replace  in struct `rpmsg_virtio_config` the `txrx_shpool` by a 
`struct rpmsg_virtio_shm_pool **txrx_shpool`,
(and add a `struct metal_io_region **shm_io `)

Reply:
Done!